src/ORM/Association/BelongsToMany.php

```diff
@@ -1002,12 +1002,12 @@ protected function _junctionTableName($name = null)
     {
         if ($name === null) {
             if (empty($this->_junctionTableName)) {
-                $aliases = array_map('\Cake\Utility\Inflector::underscore', [
-                    $this->source()->alias(),
-                    $this->target()->alias()
+                $tablesNames = array_map('\Cake\Utility\Inflector::underscore', [
+                    $this->source()->table(),
+                    $this->target()->table()
                 ]);
-                sort($aliases);
-                $this->_junctionTableName = implode('_', $aliases);
+                sort($tablesNames);
+                $this->_junctionTableName = implode('_', $tablesNames);
```

Comment:
The tests you updated were for hasMany associations which don't have junction tables, could another test be updated for this situation too?